File: 1100/2184B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long s = sc.nextLong();
            long k = sc.nextLong();
            long m = sc.nextLong();
            
            if(((m/k)&1) == 1)
                s = Math.min(s, k);
            
            System.out.println(Math.max(0, s-m%k));
        }
    }
}

File: 1100/2184B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long s, k, m;
        cin >> s >> k >> m;

        if (((m / k) & 1) == 1)
            s = min(s, k);

        cout << max(0LL, s - m % k) << "\n";
    }

    return 0;
}
